tests/test_res_diff.py

```diff
@@ -0,0 +1,238 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at:
+#
+#    http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
+# OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the
+# License.
+
+import os
+
+from tests.util import run_ion_test_driver, compare_two_files
+
+TEST_FILE_NAME = 'res_diff_tests'
+TEST_FILE_PATH = os.path.join(os.path.split(os.path.abspath(__file__))[0], TEST_FILE_NAME)
+EXPECT_FILE_NAME = 'res_diff_expect'
+EXPECT_FILE_PATH = os.path.join(os.path.split(os.path.abspath(__file__))[0], EXPECT_FILE_NAME)
+
+
+# Empty report.
+def test_empty_report():
+    test_file = 'test_empty_report.ion'
+    output_path = run_ion_test_driver(os.path.join(TEST_FILE_PATH, test_file), 'ion-java,1', 'ion-java,2')
+    res = compare_two_files(output_path, os.path.join(EXPECT_FILE_PATH, test_file))
+    os.remove(output_path)
+    assert res is True
```

Comment:
All of these tests have exactly the same structure and differ only by the test file, making them an excellent use case for a reusable function or (even better) [parameterized tests](https://docs.pytest.org/en/stable/example/parametrize.html).

ion-python has parameterized tests (e.g. [here](https://github.com/amzn/ion-python/blob/master/tests/test_vectors.py#L360)), but it uses some [custom parameterization logic](https://github.com/amzn/ion-python/blob/b6ce1defa4973825623cc95ac4be3203bbe1f69d/tests/__init__.py#L46) that you might be able to get working natively using pytest.

Reply:
Got it, created parameterized tests for each cases.